docs/rfcs/IntelGpuDialect.md

```diff
@@ -0,0 +1,122 @@
+
+## Proposal
+
+We propose to have the I-GPU dialect with just two ops called the CreateStreamOp & DestroyStreamOp.
+
+### Operation:
+
+#### intel_gpu.create_stream (intel_gpu::CreateStreamOp)
+
+Create Intel GPU Stream.
+
+Syntax:
+
+```
+operation ::= "intel_gpu.create_stream"() : () -> !intel_gpu.OpaqueType
+```
+
+The intel_gpu.create_stream() operation will be added by a pass called AddIntelGpuStream after the dialect is in Gpu Dialect. After that the IR is lowered all the way down to LLVM.
+This operation will be converted to runtime call for stream creation in the LLVM generated code. The runtime routine will create a SYCL/L0 queue, device and context.
+
+
+#### intel_gpu.destroy_stream (intel_gpu::DestroyStreamOp)
+
+Destroy Intel GPU Stream.
+
+Syntax:
+
+```
+operation ::= "intel_gpu.destroy_stream"() : () -> !intel_gpu.OpaqueType
+```
+
+The intel_gpu.destroy_stream() operation will be added by a pass called AddIntelGpuStream after the dialect is in Gpu Dialect. After that the IR is lowered all the way down to LLVM.
+This operation will destroy the above created stream.
+
+
+Example Usage of lowering:
+
+This is how the IR will look after inserting the create_stream & destroy_stream operation in the IR via a custom pass called AddIntelGpuStream.
+The create_stream op will be inserted at the begining of the IR before the device allocation ops because the device allocation requires the stream as one of the input.
+
+```
+
+func.func @main() {
+    %c8 = arith.constant 8 : index
+    %c1 = arith.constant 1 : index
+    %cst = arith.constant 2.200000e+00 : f32
+    %cst_0 = arith.constant 1.100000e+00 : f32
+    %cst_1 = arith.constant 0.000000e+00 : f32
+    %0 = "intel_gpu.create_stream"() : () -> !intel_gpu.OpaqueType
+    .
+    .
+    .
+    .
+    %6 = gpu.launch_func async [%5] @main_kernel::@main_kernel blocks in (%c8, %c1, %c1) threads in (%c1, %c1, %c1) args(%memref : memref<8xf32>, %memref_2 : memref<8xf32>, %memref_4 : memref<8xf32>)
```

Comment:
Don't user need to specify which stream to use when launch a kernel function?

Reply:
yes, if user specify the stream then we just use that, or else we will create a default one and launch kernel...